examples/platforms/nrf52840/openthread-core-nrf52840-config.h

```diff
@@ -158,6 +158,42 @@
  */
 #define OPENTHREAD_CONFIG_HEAP_SIZE_NO_DTLS                     2048
 
+/**
+ * @def OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+ *
+ * Define as 1 to enable the time synchronization service feature.
+ *
+ * @note If it's enabled, plaforms must support interrupt context and concurrent access AES.
+ *
+ */
+#define OPENTHREAD_CONFIG_ENABLE_TIME_SYNC                      0
+
+/**
+ * @def OPENTHREAD_CONFIG_TIME_SYNC_REQUIRED
+ *
+ * Define as 1 to require time synchronization when attaching to a network. If the device is router capable
+ * and cannot find a neighboring router supporting time synchronization, the device will form a new partition.
+ * If the device is not router capable, the device will remain an orphan.
+ *
+ * Applicable only if time synchronization service feature is enabled (i.e., OPENTHREAD_CONFIG_ENABLE_TIME_SYNC is set)
+ *
+ */
+#define OPENTHREAD_CONFIG_TIME_SYNC_REQUIRED                    0
+
+/**
+ * @def NRF_MBEDTLS_AES_ALT_INTERRUPT_CONTEXT
+ *
+ * Define as 1 to enable AES usage in interrupt context, by introducing a software AES under platform layer.
+ *
+ * @note This feature must be enabled when OpenThread time synchronization service is enabled.
+ *
+ */
+#ifdef OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+#define NRF_MBEDTLS_AES_ALT_INTERRUPT_CONTEXT                   1
```

Comment:
@jwhui I moved this definition to platform-config.h by the last change of #2719, sorry that I didn't verify it since I didn't expect the definitions in platform-config.h don't take effect in aes_alt.h.

But if I include platform-config.h in aes_alt.h, there are some build errors like:
```
  CXX      api/libopenthread_mtd_a-commissioner_api.o
In file included from /home/chenshu/code/openthread/examples/../third_party/NordicSemiconductor/libraries/crypto/aes_alt.h:55:0,
                 from /home/chenshu/code/openthread/examples/../third_party/mbedtls/repo.patched/include/mbedtls/aes.h:325,
                 from /home/chenshu/code/openthread/examples/../third_party/mbedtls/repo.patched/include/mbedtls/ctr_drbg.h:26,
                 from /home/chenshu/code/openthread/examples/../src/core/meshcop/dtls.hpp:42,
                 from /home/chenshu/code/openthread/examples/../src/core/coap/coap_secure.hpp:35,
                 from /home/chenshu/code/openthread/examples/../src/core/thread/thread_netif.hpp:42,
                 from /home/chenshu/code/openthread/examples/../src/core/common/instance.hpp:61,
                 from /home/chenshu/code/openthread/examples/../src/core/api/border_router_api.cpp:40:
/home/chenshu/code/openthread/examples/../examples/platforms/nrf52840/platform-config.h:40:10: fatal error: drivers/clock/nrf_drv_clock.h: No such file or directory
 #include "drivers/clock/nrf_drv_clock.h"
```

The build error is because that the header files in third_party library is not visible to OT core.

So I move this definition back here, and it works now. Please let know if you have any concern about this way, thanks.